Umee-Bots/High-Utilization-Ratio/src/agent.ts

```diff
@@ -0,0 +1,150 @@
+import BigNumber from "bignumber.js";
+import {
+  BlockEvent,
+  ethers,
+  Finding,
+  getEthersProvider,
+  HandleBlock,
+  HandleTransaction,
+  Initialize,
+  TransactionEvent,
+} from "forta-agent";
+import CONFIG from "./agent.config";
+import { GET_RESERVES_LIST_ABI, GET_RESERVE_DATA_ABI, RESERVE_INITIALIZED_ABI } from "./constants";
+import { MulticallProvider } from "./multicall";
+import {
+  AgentConfig,
+  arrayChunks,
+  createAbsoluteThresholdFinding,
+  createPercentageThresholdFinding,
+  createReserveData,
+  ReserveBalances,
+  ReserveData,
+  usageRatio,
+} from "./utils";
+
+BigNumber.set({ DECIMAL_PLACES: 18 });
+
+const reserveData: ReserveData[] = [];
+
+// testing
+export const resetReserveData = () => (reserveData.length = 0);
+export const getReserveData = () => reserveData;
+export const setReserveData = (newReserveData: ReserveData[]) => {
+  resetReserveData();
+  newReserveData.forEach((el) => reserveData.push(el));
+};
+
+let multicallProvider: MulticallProvider;
+
+export const provideInitialize = (provider: ethers.providers.Provider, config: AgentConfig): Initialize => {
+  return async () => {
+    const LendingPool = new ethers.Contract(
+      config.lendingPoolAddress,
+      [GET_RESERVES_LIST_ABI, GET_RESERVE_DATA_ABI],
+      provider
+    );
+
+    const reserveAddresses: string[] = await LendingPool.getReservesList();
+    const reserves = await Promise.all(reserveAddresses.map((el) => LendingPool.getReserveData(el)));
+
+    reserves.forEach((data, idx) => {
+      reserveData.push(
+        createReserveData(
+          reserveAddresses[idx],
+          data.uTokenAddress,
+          data.stableDebtTokenAddress,
+          data.variableDebtTokenAddress
+        )
+      );
+    });
+
+    multicallProvider = new MulticallProvider(provider);
+    await multicallProvider.init();
```

Comment:
I would move the assignment (L62) to L38 where the variable is defined and keep the `init` call here. I would add a `multicallProvider` parameter to all the `provideHandle...` so in tests you can pass the provider. 

This change along with the one for `reserveData` will allow you to remove many initialize calls you have in tests.

Reply:
Done